jumpscale/sals/process/__init__.py

```diff
@@ -84,595 +141,833 @@ def execute(
     """Check whether a given pid actually does belong to a given process name.
 
-    Arguments:
-        pid (int) -- process pid
-        process {str} -- process name
+    Args:
+        pid (int): Process ID
+        process (str): String to match againset candidate processes name using equality operator
 
     Returns:
-        [bool] -- True if process_name matched process name of the pid
+        bool: True if process_name matched process name of the pid, False otherwise.
     """
-    pid = int(pid)
-    proc = psutil.Process(pid)
-    return proc.name() == process_name
+    try:
+        proc = psutil.Process(pid)
+        return proc.name() == process_name
+    except (psutil.AccessDenied, psutil.NoSuchProcess):
+        return False
 
 
-def set_env_var(varnames, varvalues):
-    """Set the value of the environment variables C{varnames}. Existing variable are overwritten
+def set_env_var(var_names, var_values):
+    """Set the value of the environment variables {varnames}. Existing variable are overwritten
 
-    Arguments:
-        varnames {list(str)} --  A list of the names of all the environment variables to set
-        varvalues {list(str)} -- A list of all values for the environment variables
+    Args:
+        var_names list[str]: A list of the names of all the environment variables to set
+        varvalues list[str]: A list of all values for the environment variables
 
+    Raises:
+        j.exceptions.RuntimeError: if error happened during setting the environment variables
     """
     try:
-        for i in range(len(varnames)):
-            os.environ[varnames[i]] = str(varvalues[i]).strip()
+        for i in range(len(var_names)):
+            os.environ[var_names[i]] = str(var_values[i]).strip()
     except Exception as e:
         raise j.exceptions.RuntimeError(e)
 
 
-def get_pid_by_port(port):
-    """Returns pids of the process that is listening on the given port
+def get_pid_by_port(port, ipv6=False, udp=False):
+    """Returns the PID of the process that is listening on the given port
 
-    Arguments:
-        port (int) -- port number
+    Args:
+        port (int): Port number to lookup for.
+        ipv6 (bool, optional): Whether to search the connections that using ipv6 instead of ipv4. Defaults to False.
+        udp (bool, optional): Whether to search the connections for UDP port instead of TCP. Defaults to False.
 
     Returns:
-        int -- pid of process that listen on that port
+        (int or None): Process ID that listen on that port
     """
 
-    process = get_process_by_port(port)
-    if process is None:
-        return []
-    return process.pid
+    process = get_process_by_port(port, ipv6=ipv6, udp=udp)
+    if process:
+        return process.pid
 
 
-def kill_process_by_name(name, sig=signal.SIGTERM.value, match_predicate=None):
-    """Kill all processes for a given command
+def kill_process_by_name(process_name, sig=signal.SIGTERM, match_predicate=None, timeout=5, sure_kill=False):
```

Comment:
Better to use `force` or `graceful` instead of `sure_kill`?

Reply:
the same as the previous replay on a similar comment.